### PR TITLE
COG-1399 NEW: Clear Button for Search Criteria

### PR DESCRIPTION
**Pull request by sorych**
#### Which User story does this relate to (link)?
https://osi-cwds.atlassian.net/browse/COG-1399

#### Brief feature description
Clear Button for Search Criteria

### CODE REVIEW

**Comment by sorych** [comment]
WIP: I will add acceptance tests as well here

File: app/javascript/views/searchUserList/SearchUserList.test.js [hunk 1]
```diff
@@ -656,6 +677,108 @@ describe('SearchUserList', () => {
...
+          userDetails={details}
+          query={query}
+          searchPageTiles={searchPageTiles}
+        />
+      )
+      expect(component.instance().isDisabledClearBtn()).toEqual(false)
+    })
+
+    it('returns false when officeNames list is non-empty', () => {
+      const component = shallow(
```
Comment (kelfink) [inline app/javascript/views/searchUserList/SearchUserList.test.js]: This stuff gets pretty wordy.  There are some jest techniques (as in the unit test frameworks) the might let us shorten this with a for loop over the test parameters, and the 'it' block inside that, accepting a minimal set of search params to test.  It's a bit involved, though.  It's OK for now.

File: spec/acceptance/search_user_list_page_spec.rb [hunk 2]
```diff
@@ -56,8 +56,19 @@
 
   scenario 'create user button is disabled before search and enabled after' do
     login
+    undo_all_search
+    check_the_button('CREATE A NEW USER', true)
     search_users(last_name: 'Man')
-    sleep 2
```
Comment (kelfink) [inline spec/acceptance/search_user_list_page_spec.rb]: If this is reliable without the sleep, then ok.  Let's see how it does...

File: app/javascript/reducers/searchUserListReducers.js [hunk 3]
```diff
@@ -101,6 +107,14 @@ function searchUserListReducer(state = initialValue, { type, payload, error, met
         query,
       }
 
+    case actionTypes.USER_LIST_CLEAR_SEARCH:
+      return {
+        ...state,
+        inputData: initialValue.inputData,
```
Comment (kelfink) [inline app/javascript/reducers/searchUserListReducers.js]: the meat of the change.  Setting the state back to initial values (empty). :-)

File: spec/acceptance/search_user_list_page_spec.rb (hunk 2 above)
Comment (sorych) [reply spec/acceptance/search_user_list_page_spec.rb]: I don't need the search results themselves here and the buttons are updated instantly

**Comment by RamuKammagani-osi** [comment]
retest this please

**Comment by sorych** [comment]
retest this please

**Pull request merged**